marklogic-data-hub-central/ui/src/components/load/load-card.test.tsx

```diff
@@ -67,6 +67,14 @@ describe('Load Card component', () => {
     //Click on 'Yes' button
     fireEvent.click(getByTestId('testLoadXML-to-FlowA-Confirm'));
 
+    //verify Load cards are ordered by Last Updated
```

Comment:
Is this the right place to add these assertions. Previous step is clicking on Yes to add to existing flow and redirected to run tile, so this is like breaking the chain of events. This scenario probably deserves to be a separate test. You could add it to this one before load step is added to a flow, but then you also would have to update the test description.

Reply:
Okay, I will work on refactoring the test here in the same file then. Thanks!

Reply:
> Is this the right place to add these assertions. Previous step is clicking on Yes to add to existing flow and redirected to run tile, so this is like breaking the chain of events. This scenario probably deserves to be a separate test. You could add it to this one before load step is added to a flow, but then you also would have to update the test description.

@bsrikan but generally are we doing roundtrip data CRUD in the unit tests? Seems like when we test POSTs, we're just testing that the POST gets called and not going about checking the state of the existing payload for that view. I could be wrong. Is there an example similar to this? I will look...

Reply:
You can test for something like this:

```
//before update
    expect(loadCards[0]).toHaveTextContent('Add New');
    expect(loadCards[1]).toHaveTextContent('testLoadXML');
    expect(loadCards[2]).toHaveTextContent('testLoad123');
    expect(loadCards[3]).toHaveTextContent('testLoad');

//update timestamp for testLoadXML
data.loadData.data[1].lastUpdated = "1990-04-15T14:22:54.057519-07:00"

//after update
    expect(loadCards[0]).toHaveTextContent('Add New');
    expect(loadCards[2]).toHaveTextContent('testLoad123');
    expect(loadCards[3]).toHaveTextContent('testLoad');
    expect(loadCards[1]).toHaveTextContent('testLoadXML');
```

Reply:
good idea, thanks. Will work on that.

Reply:
@bsrikan, nice. We can avoid mucking with the CRUD and just change the source data and see that the UI adjusts.

Reply:
@briantang, I like @bsrikan's suggestion of making a separate test.

Reply:
Noted, will do. Thanks all :)

Reply:
@wooldridge you are right about the roundtrip. For RTL unit tests, our mock data is setting up the response for POST request. So its like testing with the assumption that server responds as expected. Now its upto the UI to render that correctly. The same is tested with scenario above. We check for the order of load cards as is. Then update the timestamp, which is to mock the edit/save load configuration, when the timestamp is updated and verify the order of load cards again.
The only roundtrip from server for this ticket is the timestamp update on edit/save. But is it worth a Cypress test, probably not IMO. If a scenario comes up in the future where we edit the settings or step details, we could additionally test for placement based on last updated.

Reply:
@wooldridge we were not able to test as suggested; rerender was not calling useEffect and so we were not able to verify the sort order after timestamp was updated within the test. Instead, the mock data itself has unordered list, so we verify that it renders the cards in the expected order.